waldo/tool/reactnative/reactnative.go

```diff
@@ -1,31 +1,256 @@
+	hasIosProject := xcode.IsPossibleXcodeContainer(iosDirPath)
+
+	return hasAndroidProject || hasIosProject
 }
 
 func MakeReactNativeBuilder(absPath, relPath string, verbose bool, ios *lib.IOStreams) (*ReactNativeBuilder, string, lib.Platform, error) {
-	return nil, "", lib.PlatformUnknown, errors.New("Don’t know how to make a React Native recipe")
+	platform, err := determinePlatform(verbose, ios)
+
+	if err != nil {
+		return nil, "", lib.PlatformUnknown, err
+	}
+
+	ios.Printf("\nFinding all supported build modes…\n")
+
+	rni, err := DetectReactNativeInfo(absPath, platform, ios)
+
+	if err != nil {
+		return nil, "", lib.PlatformUnknown, err
+	}
+
+	mode, err := determineMode(rni.Modes, verbose, ios)
+
+	if err != nil {
+		return nil, "", lib.PlatformUnknown, err
+	}
+
+	rnb := &ReactNativeBuilder{Mode: mode}
+
+	return rnb, rni.Name, platform, nil
 }
 
 //-----------------------------------------------------------------------------
 
 func (rnb *ReactNativeBuilder) Build(basePath string, platform lib.Platform, clean, verbose bool, ios *lib.IOStreams) (string, error) {
-	return "", fmt.Errorf("Don’t know how to build this app with React Native!")
+	target := rnb.formatTarget(platform)
+
+	rni, err := DetectReactNativeInfo(basePath, platform, ios)
+
+	if err != nil {
+		return "", err
+	}
+
+	ios.Printf("\nDetermining build path for %s…\n", target)
+
+	buildPath, err := rnb.determineBuildPath(basePath, rni.Name, platform, ios)
+
+	if err != nil {
+		return "", err
+	}
+
+	ios.Printf("\nBuilding %s…\n", target)
+
+	dashes := "\n" + strings.Repeat("-", 79) + "\n"
+
+	ios.Println(dashes)
+
+	if clean {
+		if err = rnb.clean(basePath, rni.Name, platform, verbose, ios); err != nil {
+			return "", err
+		}
+	}
+
+	if err = rnb.build(basePath, platform, verbose, ios); err != nil {
+		return "", err
+	}
+
+	ios.Println(dashes)
+
+	ios.Printf("\nVerifying build path for %s…\n", target)
+
+	return rnb.verifyBuildPath(buildPath, rni.Name, platform, ios)
 }
 
 func (rnb *ReactNativeBuilder) Summarize() string {
-	return ""
+	summary := ""
+
+	lib.AppendIfNotEmpty(&summary, "mode", rnb.Mode, "=", ", ")
+
+	return summary
+}
+
+//-----------------------------------------------------------------------------
+
+func (rnb *ReactNativeBuilder) androidBuildArgs() []string {
+	return []string{"build-android", "--mode", rnb.Mode, "--no-packager"}
+}
+
+func (rnb *ReactNativeBuilder) build(basePath string, platform lib.Platform, verbose bool, ios *lib.IOStreams) error {
+	args := []string{"--yes", "react-native"}
+	env := lib.CurrentEnvironment()
+
+	switch platform {
+	case lib.PlatformAndroid:
+		args = append(args, rnb.androidBuildArgs()...)
+
+	case lib.PlatformIos:
+		args = append(args, rnb.iosBuildArgs()...)
+
+		env["FORCE_BUNDLING"] = "1" // this is the magic!
+
+	default:
+		return fmt.Errorf("Unknown build platform: %q", platform)
+	}
+
+	if verbose {
+		args = append(args, "--verbose")
+	}
+
+	task := lib.NewTask("npx", args...)
```

Comment:
Better to pass `--no-install` to npx to ensure it doesn't try do do a global install of the tool if not present.

Reply:
Ah. I will add in that option.